src/mysql-grammar-2/expression/function-call/arguments.ts

```diff
@@ -20,25 +30,33 @@ export const UserDefinedExpressionList_Arguments = seq(
     field("closeParenthesesToken", TokenKind.CloseParentheses),
 );
 
-export const Expression1_Arguments = parentheses(
-    field("item", SyntaxKind.Expression),
+export const Expression1_Arguments = fieldLengthCheck(
+    "item",
+    1,
+    1,
+    SyntaxKind.ExpressionList_ArgumentsImpl
 );
 
-export const Expression2_Arguments = parentheses(seq(
-    field("item", SyntaxKind.Expression),
-    field("commaToken", itemSeparator),
-    field("item", SyntaxKind.Expression),
-));
+export const Expression2_Arguments = fieldLengthCheck(
+    "item",
+    2,
+    2,
+    SyntaxKind.ExpressionList_ArgumentsImpl
+);
 
-export const Expression4_Arguments = parentheses(seq(
-    field("item", SyntaxKind.Expression),
-    field("commaToken", itemSeparator),
-    field("item", SyntaxKind.Expression),
-    field("commaToken", itemSeparator),
-    field("item", SyntaxKind.Expression),
-    field("commaToken", itemSeparator),
-    field("item", SyntaxKind.Expression),
-));
+export const Expression4_Arguments = fieldLengthCheck(
+    "item",
+    4,
+    4,
+    SyntaxKind.ExpressionList_ArgumentsImpl
+);
+
+export const Expression1To2_Arguments = fieldLengthCheck(
+    "item",
+    1,
+    2,
+    SyntaxKind.ExpressionList_ArgumentsImpl
+);
```

Comment:
When building our grammar, we can now add field checks that don't quite count as syntax errors when the checks fail.

When field checks fail, we do not increment the errorCount for the parse tree.